### PR TITLE
WIP: 🧱 #11  Update dependencies

### PR DESCRIPTION
**Pull request by demariadaniel**
- Dependency updates for Emotion, React, Typescript, Arranger Components, and associated bug fixes

### CODE REVIEW

File: components/pages/explorer/RepoTable.tsx [hunk 1]
```diff
@@ -66,6 +66,7 @@ const getTableConfigs = ({
 			DownloadButton: {
 				customExporters,
 				downloadUrl: urlJoin(apiHost, 'download'),
+				// @ts-ignore
```
Comment (demariadaniel) [inline components/pages/explorer/RepoTable.tsx:69]: Workaround; conflict between Emotion + UiKit after update

File: pages/api/[...proxy].ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
 	// Don't forward cookies to the API:
 	req.headers.cookie = '';
 
-	console.info(`proxy without authentication - proxing to target:${target} path:${path}`);
+	console.info(`proxy without authentication - proxying to target:${target} path:${path}`);
```
Comment (demariadaniel) [inline pages/api/[...proxy].ts:38]: Spelling

File: global/utils/egoTokenUtils.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ import { EgoJwtData, UserWithId } from '../types';
 
 const { NEXT_PUBLIC_EGO_PUBLIC_KEY } = getConfig();
 
-const verifyJwt: (egoPublicKey: string) => (egoJwt?: string) => boolean =
+const verifyJwt: (egoPublicKey: string) => (egoJwt?: string) => string | boolean =
```
Comment (demariadaniel) [inline global/utils/egoTokenUtils.ts:31]: This is a TS issue; line 37 jwt.verify returns a string, `string && true` returns string

File: components/pages/explorer/RepoTable.tsx (hunk 1 above)
Comment (ciaranschutte) [reply components/pages/explorer/RepoTable.tsx:69]: @demariadaniel could you tell me a little more about this please? I'm always interested in Emotion + UIKit issues.

Comment (ciaranschutte) [reply components/pages/explorer/RepoTable.tsx:69]: Doesn't look like we're using UIKit on overture stack - unless it's a deeper depedency?

File: package.json [hunk 4]
```diff
@@ -12,42 +12,46 @@
...
 		"extra-watch-webpack-plugin": "^1.0.3",
-		"fork-ts-checker-webpack-plugin": "^8.0.0",
+		"fork-ts-checker-webpack-plugin": "^9.0.2",
 		"jest": "^27.5.1",
-		"ts-patch": "^2.1.0",
-		"typescript": "^4.7.2",
-		"typescript-transform-paths": "^3.4.6"
+		"ts-patch": "^3.1.2",
+		"typescript": "^5.5.4",
+		"typescript-transform-paths": "^3.4.7"
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply package.json:54]: Sorry I should mark this PR as WIP
Yes I agree and was working on exactly that today

[the remark this answers is not in the document]
Comment (ciaranschutte) [reply package.json:54]: Np! "draft" status for the PRs help too... if I have a couple of minutes with a cup of tea I'll take a gander at some "open" ones but will mostly avoid the "draft" ones.  There's an option in Github UI just under the "reviewers" section.

[the remark this answers is not in the document]
Comment (demariadaniel) [reply package.json:54]: I have a separate branch for Iobio Components and will create a separate branch for dependency updates

I believe there are build/transpile issues with the Iobio package and dependency updates are part of the triaging

File: components/pages/explorer/RepoTable.tsx (hunk 1 above)
Comment (demariadaniel) [reply components/pages/explorer/RepoTable.tsx:69]: This is yielding a `Type '() => EmotionJSX.Element' is not assignable to type 'ReactNode'` error

I've updated this branch so it only contains dependency updates; this error is not in develop

**Pull request closed without merging**